apps/hash-frontend/src/pages/[shortname].page/edit-pinned-entity-types-modal.tsx

```diff
@@ -0,0 +1,424 @@
+      }
+
+      items.splice(result.destination.index, 0, reorderedItem);
+
+      replace(items);
+    },
+    [fields, replace],
+  );
+
+  const handleCreateNewEntityType = useCallback(
+    async (title: string) => {
+      const { data: createdEntityType } = await createEntityType({
+        data: {
+          entityType: {
+            title,
+            type: "object",
+            properties: {},
+          },
+        },
+      });
+
+      if (createdEntityType) {
+        append(createdEntityType);
+      } else {
+        /** @todo: error handling */
+      }
+      setDisplayEntityTypesSearch(false);
+    },
+    [createEntityType, append],
+  );
+
+  const handleDiscard = useCallback(() => {
+    reset();
+    onClose();
+  }, [reset, onClose]);
+
+  const renderDraggable = useDraggableInPortal();
+
+  const isSubmitDisabled = Object.keys(formState.dirtyFields).length === 0;
+
+  return (
+    <Modal
+      {...modalProps}
+      sx={{
+        "> div": {
+          overflow: "hidden",
+          padding: 0,
+        },
+      }}
+      onClose={onClose}
+    >
+      <Box>
+        <Box
+          paddingX={3}
+          paddingBottom={1.5}
+          paddingTop={2}
+          display="flex"
+          justifyContent="space-between"
+        >
+          <Box>
+            <Typography
+              gutterBottom
+              sx={{
+                fontSize: 16,
+                fontWeight: 500,
+                color: ({ palette }) => palette.gray[80],
+              }}
+            >
+              Pinned types
+            </Typography>
+            <Typography
+              sx={{ fontSize: 14, color: ({ palette }) => palette.gray[80] }}
+            >
+              Choose up to 5 types to appear in the top-bar of your profile
+            </Typography>
+          </Box>
+          <Box>
+            <IconButton
+              onClick={onClose}
+              sx={{ marginRight: -2, marginTop: -1 }}
+            >
+              <XMarkRegularIcon />
+            </IconButton>
+          </Box>
+        </Box>
+        <Divider />
+        <Box
+          component="form"
+          onSubmit={innerSubmit}
+          padding={3}
+          paddingTop={2.25}
+        >
+          <Typography
+            variant="smallCaps"
+            sx={{ color: ({ palette }) => palette.gray[70] }}
+          >
+            Types
+          </Typography>
+          <DragDropContext onDragEnd={handleOnDragEnd}>
+            <Droppable droppableId="pinnedEntityTypes">
+              {(providedDroppable) => (
+                <Box
+                  {...providedDroppable.droppableProps}
+                  ref={providedDroppable.innerRef}
+                  sx={{
+                    marginBottom: 1,
+                    display: "flex",
+                    flexDirection: "column",
+                    rowGap: 0.5,
+                  }}
+                >
+                  {fields.map((field, index) => (
+                    <Draggable
```

Comment:
I would disable dragging / hide the drag handle if there is only one item

Reply:
Good idea, have done so in https://github.com/hashintel/hash/pull/3338/commits/bbb34228bbac17591f552354bbc1b535511e4215

Reply:
Thank you